### ISSUE TITLE
[Bug] Can not get Field content in dynamic forms

### ISSUE THREAD

**Issue by rinyuki** [salesforce/utam-js-recipes#290]
The first field in my dynamic forms is a picklist.

1. get RecordActionWrapper
2. get RecordHomeSingleColNoHeaderTemplateDesktop2
3. get Component2 by Id
4. get FieldSection2
5. get Column2 in FieldSection2.
6. get field by column2.getFieldByID(fieldID).
7. get picklist in field to set value.

### Actual behavior

**Something went wrong here**
1. Can not get Column2 from FieldSection2.
```
let column2List = await fieldSection2.getContent(Column2);
// just 2 Column2 ,but returns 9 elements
```

2. and also can not get field by Id.
```
let field = await column2.getFieldByID(fieldID);
// I try to use every item in column2List as column2, no one is right.
```

3.  Can not get element in Field.
```
let element = await field.getRecordFieldContent(Picklist) ;
// it is empty.
```

### What is your expected behavior?
My project use a dynamic forms in record layout.
I need to create a new record in my e2e test.
I want to set value in field, but I can't get element in field.

PS: I can see element in Chrome developer tool, but no in wdio.
I try to use wdio + selector to get Field, it's also empty .

### Environment

- Node.js: v18.12.1
- OS: windows
- Browser: [e.g. chrome]
- Browser Version: [e.g. 119.0.2151.97, Chrome 120.0.6099.71]

        "@wdio/appium-service": "^8.26.1",
        "@wdio/cli": "^8.26.1",
        "@wdio/local-runner": "^8.26.1",
        "@wdio/mocha-framework": "^8.24.12",
        "salesforce-pageobjects": "^5.0.2",
        "wdio-utam-service": "^2.2.0"

### Additional context

I found a URL, he had the same issuse. So I use his code as demo.
https://salesforce.stackexchange.com/questions/402434/what-am-i-missing-working-with-utam-and-creating-a-record-for-an-object-utilizin

then I print HTML to check, there are no element in records-record-picklist node.

console.log(await informationFieldSection.element.element.getHTML());

```html  {.line-numbers}
<flexipage-field-section2 lwc-mlenr16lk9-host=""><flexipage-column2 slot="columns" class="column flex-width" lwc-67tg8beuu47-host=""><fl
exipage-field data-field-id="RecordCustomerClassification_cField" lwc-6tp3buk1ht8-host=""><record_flexipage-record-field class="slds-grid slds-col slds-has-flexi-truncate" role="listitem" lw
c-618vdk5svck-host=""><records-record-picklist data-input-element-id="input-field" slot="input" lwc-2fnfn23k1jh-host=""></records-record-picklist></record_flexipage-record-field></flexipage-
field><flexipage-field data-field-id="RecordAccountName_cField" lwc-6tp3buk1ht8-host=""><record_flexipage-record-field class="slds-grid slds-col slds-has-flexi-truncate" role="listitem" lwc-
618vdk5svck-host=""><records-record-layout-lookup data-input-element-id="input-field" slot="input" lwc-5tti5094td7-host=""></records-record-layout-lookup></record_flexipage-record-field></fl
exipage-field><flexipage-field data-field-id="RecordAccountManage_cField" lwc-6tp3buk1ht8-host=""><record_flexipage-record-field class="slds-grid slds-col slds-has-flexi-truncate" role="list
item" lwc-618vdk5svck-host=""><records-record-layout-lookup data-input-element-id="input-field" slot="input" lwc-5tti5094td7-host=""></records-record-layout-lookup></record_flexipage-record-
field></flexipage-field></flexipage-column2><flexipage-column2 slot="columns" class="column flex-width" lwc-67tg8beuu47-host=""><flexipage-field data-field-id="RecordTargetReport_cField3" lw
c-6tp3buk1ht8-host=""><record_flexipage-record-field class="slds-grid slds-col slds-has-flexi-truncate" role="listitem" lwc-618vdk5svck-host=""><records-record-layout-checkbox data-input-ele
ment-id="input-field" slot="input" lwc-gjkarqdli1-host=""></records-record-layout-checkbox></record_flexipage-record-field></flexipage-field></flexipage-column2></flexipage-field-section2>
```

in chrome developer tool, I can see a lot element in records-record-picklist(lightning-picklist, lightning-combox...)

**Comment by lizaiv77** [comment]
@rinyuki 
1. for `let column2List = await fieldSection2.getContent(Column2);` Is fieldSection2 a page object utam/flexipage/fieldSection2? is Column2 utam/flexipage/column2?
2. HTML markup snippet you attached is incomplete, for start it does not have opening tag `flexipage-field-section2`
3. For `let field = await column2.getFieldByID(fieldID);` Not sure what you mean. So did you get to column2 and method getFieldByID did not work? What was the error?
4. `let element = await field.getRecordFieldContent(Picklist);` Empty as in null? How you got to field? What PO is Picklist? Is HTML same as on snippet?

**Comment by rinyuki** [comment]
> @rinyuki
> 
> 1. for `let column2List = await fieldSection2.getContent(Column2);` Is fieldSection2 a page object utam/flexipage/fieldSection2? is Column2 utam/flexipage/column2?
> 2. HTML markup snippet you attached is incomplete, for start it does not have opening tag `flexipage-field-section2`
> 3. For `let field = await column2.getFieldByID(fieldID);` Not sure what you mean. So did you get to column2 and method getFieldByID did not work? What was the error?
> 4. `let element = await field.getRecordFieldContent(Picklist);` Empty as in null? How you got to field? What PO is Picklist? Is HTML same as on snippet?

Thank you for comment.
1. yes, and I think my definition should be correct.
2. sorry, I had updated HTML.
3. I try to explant my flow by code ⇩.
4. The field that I want to set value is a picklist. But faild from step 1,so 2,3,4 also not work as I wanted.
```
import RecordHomeSingleColNoHeaderTemplateDesktop2 from "salesforce-pageobjects/flexipage/pageObjects/recordHomeSingleColNoHeaderTemplateDesktop2";
import FieldSection2 from "salesforce-pageobjects/flexipage/pageObjects/fieldSection2";
import Column2 from "salesforce-pageobjects/flexipage/pageObjects/column2";
import Field from "salesforce-pageobjects/flexipage/pageObjects/field";
import RecordField from "salesforce-pageobjects/record/flexipage/pageObjects/recordField";
import Picklist from "salesforce-pageobjects/lightning/pageObjects/picklist";

// main
const recordActionWrapper = await utam.load(RecordActionWrapper);
    const dynamicFormFlexipage =
      await recordActionWrapper.waitForRecordHomeSingleColNoHeaderTemplateDesktop2();
// get component2 from Id, and get FieldSection2
    const informationFieldSection = await (
      await dynamicFormFlexipage.getComponent2("flexipage_fieldSection3")
    ).getContent(FieldSection2);
// get HTML, no node in tag records-record-layout-lookup
    console.log(await informationFieldSection.element.element.getHTML());
// get column2
    const informationFieldSectionColumns =
      await informationFieldSection.getContent(Column2);
// get fields, await column2.getFieldByID(fieldID); not work
    const fields = await informationFieldSectionColumns[0].getFields(Field);
    const field = fields[0];
// stop here
    await field.waitForRecordFieldContent();
    const fieldInput = await field.getRecordFieldContent(RecordField);
    const fieldLightningInput = await fieldInput.getFieldContentEditMode(
      RecordLayoutBaseInput
    );
```

**Comment by lizaiv77** [comment]
Sorry for delayed response.
So from what I understand code `informationFieldSection.getContent(Column2)` returns 9 elements instead of 2, correct?  It's clear from HTML that there are only 2 columns, so you're likely in wrong section. Which also explains why `getFieldByID(fieldID)` does not work.

Unfortunately the only thing I can advise without virtual meeting is to check in browser console that methods you're calling to get to fieldSection2 producing correct selectors, see if it finds the right element on the page. 
```
querySelector(“.oneRecordActionWrapper”).querySelector(“flexipage-record-home-single-col-no-header-template-desktop2”).querySelector(“flexipage-component2[data-component-id=‘flexipage_fieldSection3’]”).querySelector(“:scope *”)
```

**Comment by rinyuki** [comment]
@lizaiv77 

Thank you for response.

I had the same idea and verified that fieldSection2 is correct.
The code you provided and the result is also correct.

>> informationFieldSection.getContent(Column2) returns 9 elements instead of 2

I have confirmed the return value. Among the 9 return values, some return the Column2 node, some return only the record_flexipage-record-field node, and some only return the records-record-picklist node. I mean, many combinations are returned.

Although the returned value contains the Column2 node, I cannot use it. `getFieldByID(fieldID)`.

The result confirmation of this code is weird, I only found one person who had the same problem as me (URL above).
I'd also be tempted to use video conferencing to reproduce the issue, but that would probably get me fired.
I think this should be a wdio BUG.


Because Puppeteer in wdio got the node and included everything.
just like `let Puppeteer = await browser.getPuppeteer();`
